File: SooYeon/231127_week/Programmers_42747_H-index.cpp
#include <bits/stdc++.h>
using namespace std;

bool compare(int a, int b) {
    return a > b;
}

int solution(vector<int> citations) {
    // 인용된 논문 개수
    int answer = 0;
    
    // 내림차순으로 인용 횟수 정렬
    sort(citations.begin(), citations.end(), compare);
    
    // 논문 인용 횟수가 인용된 논문 개수와 같거나 적어지면 break
    // 인용된 논문 개수가 논문 인용 횟수보다 커지는 순간, 문제에서 원하는 h편 이상을 적용할 수 없음
    for(int i = 0; i < citations.size(); i++) {
        if(answer >= citations[i]) break;
        answer++;
    };
    
    return answer;
}

File: MinJu/231204_week/Programmers_42747_H-Index.py
def solution(citations):
    answer = len(citations)
    citations.sort(reverse=True) # 내림차순 
    
    #H_index가 존재하고 넘는 논문의 개수 
    for i in range(len(citations)):
        if(citations[i] < i+1):
            return i
    return answer # 횟수가 같으면 전체 